src/librustc_typeck/diagnostics.rs

```diff
@@ -2109,6 +2109,41 @@ E0380: r##"
 Default impls are only allowed for traits with no methods or associated items.
 For more information see the [opt-in builtin traits RFC](https://github.com/rust
 -lang/rfcs/blob/master/text/0019-opt-in-builtin-traits.md).
+"##,
+
+E0392: r##"
+This error indicates that a type parameter has been declared but not actually
+used.
+
+Here is an example that demonstrates the error:
+
+```
+enum Foo<T> {
+    Bar
+}
+```
+
+The first way to fix this error is by removing the type parameter, as
+shown below:
+
+```
+enum Foo {
+    Bar
+}
+```
+
+The second method is to actually make use of the type parameter:
+
+```
+enum Foo<T> {
+    Bar(T)
+}
+```
+
```

Comment:
In the second method, if you don't want to use the type parameter but need it to exist as a type safety thing, you're supposed to use a `PhantomData` (see the error note that comes with this error) as a marker type.

http://doc.rust-lang.org/std/marker/struct.PhantomData.html